### ISSUE TITLE
Loss of distillation is not sampled from the teacher generation, but from the training data response

### ISSUE THREAD

**Issue by liuxy1103** [shufangxun/LLaVA-MoD#8]
![image](https://github.com/user-attachments/assets/2ffa0337-95bc-438f-a03b-738fc6cc434f)

        policy_logprobs, policy_sft_loss, policy_moe_loss, policy_labels = self.get_logp(model, inputs)

        align_loss = self.compute_align_loss(policy_logprobs=policy_logprobs,
                                             reference_probs=reference_probs,
                                             labels=policy_labels)  
                                             
As you can see from the code, the loss of distillation is not sampled from the teacher generation, but from the training data response

**Comment by shufangxun** [comment]
> ![image](https://github.com/user-attachments/assets/2ffa0337-95bc-438f-a03b-738fc6cc434f)
> 
> ```
>     policy_logprobs, policy_sft_loss, policy_moe_loss, policy_labels = self.get_logp(model, inputs)
> 
>     align_loss = self.compute_align_loss(policy_logprobs=policy_logprobs,
>                                          reference_probs=reference_probs,
>                                          labels=policy_labels)  
> ```
> 
> As you can see from the code, the loss of distillation is not sampled from the teacher generation, but from the training data response

Hello, we sample the response logits of training data from teacher model.
